xl2times/datatypes.py

```diff
@@ -70,6 +71,55 @@ def has_tag(cls, tag):
         return tag in cls._value2member_map_
 
 
+class DataModule(str, Enum):
+    """Categorise data into modules based on the file they are coming from."""
+
+    base = "VT_*.*, BY_Trans.*"
+    syssettings = "SysSettings.*"
+    subres = "SubRES_TMPL/SubRES_*.*"
+    sets = "Set*.*"
+    lma = "LMA*.*"
+    demand = "SuppXLS/Demands/Dem_Alloc+Series.*, SuppXLS/Demands/ScenDem_*.*"
+    scen = "SuppXLS/Scen_*.*"
+    trade = "SuppXLS/Trades/ScenTrade_*.*"
+
+    @classmethod
+    def is_module_type(cls, path):
+        for data_module in cls:
+            if any(
+                PurePath(path.lower()).match(pattern.lower().strip())
+                for pattern in data_module.value.split(",")
+            ):
+                return data_module
+        return None
+
+    @classmethod
+    def is_submodule_type(cls, path):
+        match cls.is_module_type(path):
+            case DataModule.base | DataModule.subres:
+                if PurePath(path.lower()).match("*_trans.*"):
+                    return "trans"
+                else:
+                    return "main"
+            case None:
+                return None
+            case _:
+                return "main"
+
+    @classmethod
+    def is_module_name(cls, path):
```

Comment:
I'm used to function names beginning `is_*` to be those that return True/False, but I think all 3 of the above methods don't return booleans? Perhaps it's clearer to call them `module_type`, etc?

Reply:
Perhaps helpful in your next revision to also include type signatures to these methods. 🙏

Reply:
How about now?